app/server/src/main/scala/org/bitcoins/server/Main.scala

```diff
@@ -165,12 +165,19 @@ object Main extends App {
       wallet.processBlock(block)
       ()
     }
+    lazy val onHeaders: OnBlockHeadersReceived = { headers =>
+      wallet.updateTxoStatesFromHeader(headers.last)
```

Comment:
Since we will only be updating based off height it only makes sense to grab the header with the largest height.

As for the async issues, the others do the same. I think it is okay because it is just something called for the wallet, the `Unit` return is a safety measure so the node doesn't receive any data that it doesn't want/need (at least that's what it seems like to me)

Reply:
>Since we will only be updating based off height it only makes sense to grab the header with the largest height.

First off, this is an implementation detail that I definitely don't think we should rely on long term (reorgs, heaviest POW chain etc). 

Also do  we have any sort of guarantee that our internal processing of headers in the chain project is finished before this callback is called? If I remember correctly, for neutrino to update a transaction we need to: 

1. Get the block header
2. Get the compact filter header
3. Get the compact filter
4. See if any of our utxos match the filter 
5. Fetch the full block
6. Scan the full block for utxos
7. Update the the `SpendingInfoTable`

Also it seems like just a bad logical break to be ignoring _all_ headers we see except the last one. Our wallet should be built to update when we receive a single block, not a `Vector`. It's too hard to reason about when we start thinking about updating receiving sequences of blocks at the same time. We should iterate through each of those single blocks and apply the appropriate updates _for that single block_ to the wallet, and then move onto the next one. This callback breaks that. 

>As for the async issues, the others do the same. I think it is okay because it is just something called for the wallet, the Unit return is a safety measure so the node doesn't receive any data that it doesn't want/need (at least that's what it seems like to me)

Yeah, I don't like how we handle that to be honest though as it isn't really safe to use for exactly this reason. How do you know if your callback is done processing? We use async code everywhere internally.

Reply:
Yes. We'll need to change callbacks to return types to `Future[]`, and use the returned values from call backs in `for`/`flatMap` statements to prevent concurrency issues.